tracing-subscriber/src/registry/fmt.rs

```diff
@@ -0,0 +1,349 @@
+        }
+    }
+}
+
+// This, like the MakeWriter block, needs to be a seperate impl block because we're
+// overriding the `E` type parameter with `E2`.
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Sets a [FormatEvent<S, N>].
+    pub fn with_event_formatter<E2>(self, e: E2) -> FmtLayerBuilder<S, N, E2, W> {
+        FmtLayerBuilder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: e,
+            make_writer: self.make_writer,
+            is_interested: self.is_interested,
+            _inner: self._inner,
+        }
+    }
+}
+
+// this needs to be a seperate impl block because we're re-assigning the the W2 (make_writer)
+// type paramater from the default.
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W> {
+    /// Sets a [MakeWriter] for spans and events.
+    pub fn with_writer<W2>(self, make_writer: W2) -> FmtLayerBuilder<S, N, E, W2>
+    where
+        W2: MakeWriter + 'static,
+    {
+        FmtLayerBuilder {
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            is_interested: self.is_interested,
+            make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> FmtLayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Builds a [FmtLayer] infalliably.
+    pub fn build(self) -> FmtLayer<S, N, E, W> {
+        FmtLayer {
+            is_interested: self.is_interested,
+            make_writer: self.make_writer,
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl Default for FmtLayerBuilder {
+    fn default() -> Self {
+        Self {
+            is_interested: Box::new(|_| true),
+            fmt_fields: format::DefaultFields::default(),
+            fmt_event: format::Format::default(),
+            make_writer: io::stdout,
+            _inner: PhantomData,
+        }
+    }
+}
+
+impl<S, N, E, W> FmtLayer<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a> + LookupMetadata,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    #[inline]
+    fn make_ctx<'a>(&'a self, ctx: Context<'a, S>) -> FmtContext<'a, S, N> {
+        FmtContext {
+            ctx,
+            fmt_fields: &self.fmt_fields,
+        }
+    }
+}
+
+// === impl Formatter ===
+
+/// A newtype for storing formatted fields in a span's extensions.
+///
+/// By storing [FormattedFields] instead of a [String] directly,
+/// [FmtLayer] is able to be more defensive about other layers
+/// accidentally a span's extensions.
```

Comment:
How about rephrasing this as something like
> Because `FormattedFields` is generic over the type of the formatter that produced it, multiple versions of a span's formatted fields can be stored in the [`Extensions`] type-map. This means that when multiple formatters are in use, each can store its own formatted representation without conflicting.

or similar?